webpack.config.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Use `npm run webpack` to produce Webpack bundle for this library.
+
+const path = require('path');
+
+module.exports = {
+  entry: './src/index.ts',
+  resolve: {
+    extensions: ['.ts', '.js', '.json'],
+    alias: {
+      '../../package.json': path.resolve(__dirname, 'package.json'),
+    },
+  },
+  output: {
+    library: 'GoogleApis',
+    filename: 'googleapis.min.js',
+    path: path.resolve(__dirname, 'dist'),
+  },
+  node: {
+    child_process: 'empty',
+    fs: 'empty',
+    crypto: 'empty',
+  },
+  module: {
+    rules: [
+      {
+        test: /src\/crypto\/node\/crypto/,
+        use: 'null-loader',
+      },
+      {
+        test: /node_modules\/https-proxy-agent\//,
```

Comment:
It sucks that you need to call this out both here and in google-auth-library.  Is there a way to extend the config here for users so they don't have to suffer this?

Reply:
I will see if we can do something. One obvious way is to publish our webpack config to npm and require it from `node_modules`.

Actually, we do **not** need to have any webpack config in the auth library; it's there only for those who want to have a bundle of auth library. Webpack does not read webpack configs of the dependencies, that is, when we webpack the client library, it won't read auth library's webpack config. Hope I made it clear (this comment contains a word 'webpack' 7 times and 'config' 5 times which probably is more than enough).

Reply:
I think I understand.  So you're saying that the end users of a library need to figure out on their own which dependencies need to be ignored the whole way down through their transitive dependency tree?

Reply:
Yes, that's what I figured. And this is logical. Assume someone wants to use crypto polyfills that are provided by webpack by default; they won't be happy with our decision to strip them.

If a user just needs to use our library (e.g. drive) in the browser app, they use our published webpack config to build a bundle as we suggest. If they want to webpack their own Node.js application that depends on our client library (in terms of `package.json` `dependencies`), it's their job to write the config (they can reuse the one we use, or decide to include some of the modules we removed).

Reply:
(our part of the job here is to make sure the __code__ works - it does; everything related to the bundle size and composition should be taken care of by a person who's making the bundle)

Reply:
SGTM.